smtpd/mail.lmtp.c

```diff
@@ -209,17 +209,42 @@ lmtp_connect(const char *destination)
 }
 
 static void
-lmtp_engine(FILE *conn, struct session *session)
+lmtp_engine(int fd_read, struct session *session)
 {
+	int fd_write = 0;
+	FILE *file_read = 0;
+	FILE *file_write = 0;
 	char *line = NULL;
 	size_t linesize = 0;
 	ssize_t linelen;
 	enum phase phase = PHASE_BANNER;
 
+	fd_write = dup(fd_read);
+
+	if (fd_write == -1){
+		err(EX_TEMPFAIL, "dup");
+		return;
```

Comment:
err() doesn't return, so this can be replaced with a single:

if ((fd_write = dup(fd_read)) == -1)
    err(EX_TEMPFAIL, "dup"):